### PR TITLE
Bugfix/sl 1115/session timeout

### PR DESCRIPTION
**Pull request by leszekgu**

### CODE REVIEW

File: src/app/security/services/auth-http-interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpResponse } from '@angular/common/http';
+import { AuthorizationHeaderName } from '../models/access-token';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { catchError, tap } from 'rxjs/operators';
+import 'rxjs/add/observable/throw';
+import { SecurityContext } from './security-context.service';
+import { Router } from '@angular/router';
+
+const new_token_header_name = 'Refreshed-Token';
```
Comment (zielinskikamil) [inline src/app/security/services/auth-http-interceptor.ts]: can you rename it to refreshRokenHeader ?

File: src/app/security/services/auth-http-interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return next.handle(req).pipe(
+            tap(event => {
+                if (event instanceof HttpResponse) {
+                    const newToken = event.headers.get(new_token_header_name);
+                    if (newToken) {
+                        this.security.setToken(newToken);
+                    }
+                }
+            }),
+            catchError((errorResponse, c2) => {
```
Comment (zielinskikamil) [inline src/app/security/services/auth-http-interceptor.ts]: what c2 means?

File: src/app/security/services/auth-http-interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        req = this.addAuthenticationHeader(req);
+        return next.handle(req).pipe(
+            tap(event => {
+                if (event instanceof HttpResponse) {
+                    const newToken = event.headers.get(new_token_header_name);
+                    if (newToken) {
+                        this.security.setToken(newToken);
```
[the remark this answers is not in the document]
Comment (leszekgu) [reply src/app/security/services/auth-http-interceptor.ts]: yes, refresh is maybe not the best word, but the new token with new jwt expiry is returned, but with still the same maxEpiryDate included.
The first is for jwt checks, the latter is for our own.

File: src/app/security/services/security.service.spec.ts [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
 import { SecurityService } from './security.service';
-import { AuthorizationHeaderName } from '../models/access-token';
 import { User } from '../models/user.model';
+import { SecurityContext } from './security-context.service';
+import { TestBed } from '@angular/core/testing';
+import { AuthorizationHeaderName } from '../models/access-token';
 
-let storageSpy, httpMock, securityService: SecurityService;
+let storageSpy, httpMock, securityService: SecurityService, securityContext: SecurityContext;
+const fakeUrl = 'https://doesnot.matter.com';
```
Comment (kpakur) [inline src/app/security/services/security.service.spec.ts]: do you know that this url actually exists?, could not you use something form platform.hmcts.net domain?

**Pull request merged**